### ISSUE TITLE
Use as Python library

### ISSUE THREAD

**Issue by frafra** [dalibo/ldap2pg#355]
It would be nice to have a function to call ldap2pg synchronization which does not try to parse `sys.args` and uses a dict as configuration instead of a file, so that ldap2pg can be used inside another Python program, without relying on `subprocess`.

**Comment by bersace** [comment]
Hi @frafra, thanks for reaching !

Interesting. Can you detail me some use case ?

By `dict`, you me the actual YAML content as parsed by pyyaml ?

How would you handle envvars ?

Regards,
Étienne

**Comment by frafra** [comment]
Sure :) I have a program which setups and configure Postgres databases, run some routines, including executing ldap2pg based on a template configuration file. I use suprocess for that at the moment, but it would be nicer to call ldap2pg from Python.

> By dict, you me the actual YAML content as parsed by pyyaml ?
Yes.

> How would you handle envvars ?
As a **kwargs or as a function argument?

**Comment by bersace** [comment]
Ok, I suggest an API like this:

``` python
def synchronize(config: dict, environ: dict=None, args: list=None):
    ...
```

You could use it like:

``` python
synchronize(config, args=["--dry-run"])
```

This would raise regular exception on failure. Logging won't be configured and nor sys.exit be called.

What do you think of this ?

**Comment by frafra** [comment]
That would be great :)

**Comment by bersace** [comment]
Available just now in ldpa2pg 5.6 :-)
